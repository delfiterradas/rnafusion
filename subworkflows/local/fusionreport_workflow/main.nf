include { FUSIONREPORT_DETECT      }     from '../../../modules/nf-core/fusionreport/detect/main'


workflow FUSIONREPORT_WORKFLOW {
    take:
        fusionreport_ref
        arriba_fusions
        starfusion_fusions
        fusioncatcher_fusions

    main:
        ch_versions = Channel.empty()
        ch_report = Channel.empty()
        ch_csv = Channel.empty()

        def ch_fusions = arriba_fusions
            .join(starfusion_fusions, failOnMismatch:true, failOnDuplicate:true)
            .join(fusioncatcher_fusions, failOnMismatch:true, failOnDuplicate:true)

        FUSIONREPORT_DETECT(ch_fusions, fusionreport_ref, params.tools_cutoff)
        ch_fusion_list = FUSIONREPORT_DETECT.out.fusion_list
        ch_fusion_list_filtered = FUSIONREPORT_DETECT.out.fusion_list_filtered
        ch_versions = ch_versions.mix(FUSIONREPORT_DETECT.out.versions)
        ch_report = FUSIONREPORT_DETECT.out.report
        ch_csv = FUSIONREPORT_DETECT.out.csv

    emit:
        versions                 = ch_versions
        fusion_list              = ch_fusion_list
        fusion_list_filtered     = ch_fusion_list_filtered
        report                   = ch_report
        csv                      = ch_csv

}
