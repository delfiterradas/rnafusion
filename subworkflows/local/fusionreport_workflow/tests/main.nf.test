nextflow_workflow {

    name "Test Subworkflow FUSIONREPORT_WORKFLOW"
    script "../main.nf"
    workflow "FUSIONREPORT_WORKFLOW"

    tag "subworkflow"
    tag "fusionreport_workflow"
    tag "fusionreport"

    test("FUSIONREPORT_WORKFLOW - Full Test") {

        setup {
            run("FUSIONREPORT_DOWNLOAD") {
                script "../../../../modules/local/fusionreport/download/main.nf"
                process {
                    """
                    """
                }
            }
        }

        when {
            workflow {
                """
                // Input channels
                input[0] = Channel.of(
                    [ [ id:'test_sample' ],
                    file("https://github.com/nf-core/test-datasets/raw/rnafusion/testdata/human/reads_1.fq.gz") ]
                )

                input[1] = FUSIONREPORT_DOWNLOAD.out.fusionreport_ref

                input[2] = Channel.of(
                    [[ id:'test_sample' ],
                        file("https://github.com/Clinical-Genomics/fusion-report/raw/master/tests/test_data/arriba.tsv") ]
                )

                input[3] =  Channel.of(
                    [[ id:'test_sample' ],
                        file("https://github.com/Clinical-Genomics/fusion-report/raw/master/tests/test_data/starfusion.tsv") ]
                )

                input[4] =  Channel.of(
                    [[ id:'test_sample' ],
                        file("https://github.com/Clinical-Genomics/fusion-report/raw/master/tests/test_data/fusioncatcher.txt") ]
                )

                """
            }
            params {
                fusioninspector_only = false
                tools_cutoff = 1
                arriba          = true
                starfusion      = true
                fusioncatcher   = true
                no_cosmic       = true
                outdir               = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.fusion_list,
                    workflow.out.fusion_list_filtered,
                    workflow.out.report,
                    workflow.out.csv,
                    workflow.out.versions
                ).match() }
            )
        }
    }


}
