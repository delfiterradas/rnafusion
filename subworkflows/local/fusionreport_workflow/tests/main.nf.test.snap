{
    "FUSIONREPORT_WORKFLOW - Full Test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.fusionreport.tsv:md5,3593b7021f26cc5427fdc96f0d1c72f0"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.fusionreport_filtered.tsv:md5,3593b7021f26cc5427fdc96f0d1c72f0"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample_fusionreport_index.html:md5,3513bcaa58446399c0957db69402d3bd"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.fusions.csv:md5,49f378c2112d7e0b3b17d9095c79e6bd"
                ]
            ],
            [
                "versions.yml:md5,90749dbf8e3e7b259c935eabb8c6ce1e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-07T13:31:41.215356596"
    }
}